src/server-glue/webresource-handlers/S3Handler.ts

```diff
@@ -0,0 +1,105 @@
+import { optionalVar, requiredVar } from '@balena/env-parsing';
+import {
+	IncomingFile,
+	UploadResponse,
+	WebResourceHandler,
+} from '../webresource-handler';
+import {
+	S3Client,
+	S3ClientConfig,
+	DeleteObjectCommand,
+} from '@aws-sdk/client-s3';
+import { Upload } from '@aws-sdk/lib-storage';
+import { v4 as uuidv4 } from 'uuid';
+
+export interface S3HandlerConfig {
+	bucket?: string;
+	region?: string;
+	accessKeyId?: string;
+	secretAccessKey?: string;
+	endpoint?: string;
+}
+
+export class S3Handler implements WebResourceHandler {
+	private readonly bucket: string;
+	private readonly region: string;
+	private readonly accessKeyId: string;
+	private readonly secretAccessKey: string;
+	private readonly endpoint: string;
+	private client: S3Client;
+
+	constructor(config?: S3HandlerConfig) {
+		this.bucket =
+			config?.bucket ??
+			optionalVar('S3_STORAGE_ADAPTER_BUCKET', 'balena-pine-web-resources');
+		this.region =
+			config?.region ?? optionalVar('S3_STORAGE_ADAPTER_REGION', 'us-east-1');
+		this.accessKeyId = config?.accessKeyId ?? requiredVar('S3_ACCESS_KEY');
+		this.secretAccessKey =
+			config?.secretAccessKey ?? requiredVar('S3_SECRET_KEY');
+		this.endpoint = config?.endpoint ?? requiredVar('S3_ENDPOINT');
+
+		this.client = this.getS3Client();
+	}
+
+	public async handleFile(resource: IncomingFile): Promise<UploadResponse> {
+		let size = 0;
+		const key = `${resource.fieldname}_${uuidv4()}_${resource.originalname}`;
+		const params = {
+			Bucket: this.bucket,
+			ACL: 'public-read',
+			StorageClass: 'STANDARD',
```

Comment:
Is this a configuration or a fixed usage of the bucker params? If this is the only way the file to bucket works, the bucket needs to be configured this way, right? Please add this in the readme then.